### PR TITLE
Feature/brickiterator

### PR DESCRIPTION
**Pull request by martinfalk**
iterator for iterating over a brick (or subvolume) within 3D data, e.g. a volume.

### CODE REVIEW

**Review by petersteneteg — APPROVED** [review]

File: include/inviwo/core/util/indexmapper.h [hunk 1]
```diff
@@ -54,7 +54,7 @@ struct IndexMapper<2, IndexType> {
     }
 
 private:
-    const IndexType dimx;
+    IndexType dimx;
```
Comment (r-englund) [inline include/inviwo/core/util/indexmapper.h:57]: Why remove const here?

Comment (petersteneteg) [reply include/inviwo/core/util/indexmapper.h:57]: To allow the assignment operator to work

Comment (r-englund) [reply include/inviwo/core/util/indexmapper.h:57]: But, there are no assignment operator :P

**Review by r-englund — CHANGES_REQUESTED** [review]
Needs documentation on what it is and how to use it.  (as we discussed IRL)

**Review by r-englund — APPROVED** [review]

**Pull request merged**